pkg/util/selfLink.go

```diff
@@ -0,0 +1,11 @@
+package util
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func GetSelfLink(obj unstructured.Unstructured) string {
```

Comment:
Do we want to just call this "GetUniqueKey" or similar? So that we aren't as implicitly coupled to the API Server's URL scheme?

Reply:
updated with GetUniqueKey